src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
-export interface Figure {
+type Colors = 'red' | 'green' | 'blue';
+type Shapes = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  color: Colors
+  shape: Shapes
+  getArea: () => {}
 }
 
-export class Triangle {
+const rounded = (number: number): number => {
+  return Math.floor(number * 100) / 100;
+};
 
-}
+export class Triangle implements Figure {
+  public shape: Shapes = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
 
-export class Circle {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('side cannot be <= 0');
+    }
 
+    const longestSide = Math.max(a, b, c);
+    const oppositeSides = (a + b + c) - longestSide;
+
+    if (longestSide >= oppositeSides) {
+      throw new Error('Not a triangle!');
+    }
+  }
+
+  getArea(): number {
+    const s = 0.5 * (this.a + this.b + this.c);
```

Comment:
what is 's' ?)

Reply:
This is the method of getting the area of a triangle. https://en.wikipedia.org/wiki/Heron's_formula